controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -182,6 +196,71 @@ func (r *RegisteredClusterReconciler) Reconcile(computeContextOri context.Contex
 	return ctrl.Result{}, nil
 }
 
+func (r *RegisteredClusterReconciler) checkSynctargetExists(locationContext context.Context, regCluster *singaporev1alpha1.RegisteredCluster) (int, error) {
+
+	locationClusterName, _ := logicalcluster.ClusterFromContext(locationContext)
+	labels := RegisteredClusterNamelabel + "=" + regCluster.Name + "," + RegisteredClusterNamespacelabel + "=" + regCluster.Namespace + "," + RegisteredClusterWorkspace + "=" + strings.ReplaceAll(locationClusterName.String(), ":", "-")
```

Comment:
This looks like the wrong label for RegisteredClusterWorkspace... I assume you want the workspace where the RegisteredCluster is located, but looks like you're using the location workspace, right? Please double check.

Shall we also use the RegisteredClusterUidLabel to uniquely identify the registered cluster?

Reply:
I'll add RegisteredClusterUidLabel label too